ra/ra.go

```diff
@@ -1915,6 +1929,10 @@ func (ra *RegistrationAuthorityImpl) createPendingAuthz(ctx context.Context, reg
 		Status:         &status,
 		Expires:        &expires,
 	}
+	if features.Enabled(features.NewAuthorizationSchema) {
```

Comment:
minor nit: you check this feature flag twice and also need a `true` bool when its enabled for this part of the code.
You could simplify this by having `v2 := features.Enabled(features.NewAuthorizationSchema)` near the top of the function. Afterwards the conditional on L1891 could be changed to `if v2 {` and this conditional assignment of `authz.v2` and the throw-away `v2` var could be replaced with an unconditional `authz.V2 = &v2`.